pkg/controller/snapshot_controller.go

```diff
@@ -682,42 +705,22 @@ func (ctrl *csiSnapshotController) updateSnapshotStatus(snapshot *crdv1.VolumeSn
 			status.Error = nil
 			change = true
 		}
-		if status.CreationTime == nil {
-			status.CreationTime = timeAt
-			change = true
-		}
 	}
-
-	/* TODO FIXME
-	switch csistatus.Type {
-	case csi.SnapshotStatus_READY:
-		if bound {
-			status.Ready = true
-			// Remove the error if checking snapshot is already bound and ready
-			status.Error = nil
-			change = true
+	if inErr != nil {
```

Comment:
I don't think we need to pass inErr into the function. There are two places this function updateSnapshotStatus is called. If there is error during createSnapshotCall, we have a separate logic to update error status.